core/server/api_container/server/startosis_engine/plan_yaml/plan_yaml_generator.go

```diff
@@ -375,6 +397,20 @@ func (planYaml *PlanYaml) RemoveService(serviceName string) {
 	}
 }
 
+func (planYaml *PlanYamlGenerator) AddPackageDependencies(packageDependency map[string]bool) {
+	for dependency := range packageDependency {
+		planYaml.privatePlanYaml.PackageDependencies = append(planYaml.privatePlanYaml.PackageDependencies, dependency)
+	}
+	slices.Sort(planYaml.privatePlanYaml.PackageDependencies)
+}
+
+func (planYaml *PlanYamlGenerator) AddImages() {
```

Comment:
I'm puzzled about the usage of this no-args `AddImages`. I would have expected images to be a params here, same as `AddPackageDependencies`. If they both have params, my mental model of the PlanYamlGenerator feels complete: "it's a builder for the plan YAML, where I chonk things into it and they get serialized out when I press 'generate'".

With `AddImages` not having params, I'm confused because it seems like there's this hidden `planYaml` private variable where images somehow get populated (but not sure how that is). That would suggest to me that the PlanYamlGenerator is actually not the space where I chonk in the data I want the plan YAML to have, but that I need to initialize it at creation with the data I want it to have (and the data-chonking happens elsewhere), but I don't see a constructor that instantiates a generator from a preexisting `planYaml` object (and that feels like it'd be more complex to me than the model of the PlanYamlGenerator as a builder).

Reply:
removed `AddImages`. Its not needed as `addImage` handles accumulating all images into the image map set and which gets converted to a list when the plan yaml is generated